### PR TITLE
Fixes anonymous field from supporter on payment details

### PR DESCRIPTION
**Pull request by clarissalimab**
Closes https://github.com/CommitChange/tix/issues/3524

### CODE REVIEW

File: app/views/nonprofits/payments/_details_table.html.erb [hunk 1]
```diff
@@ -255,7 +255,7 @@
 		</tr>
     <tr>
       <td>Anonymous?</td>
-      <td><!--= if payment_details.data.supporter.anonymous (put 'Yes') (put 'No') --></td>
+      <td><!--= if payment_details.data.anonymous_supporter (put 'Yes') (put 'No') --></td>
```
Comment (clarissalimab) [inline app/views/nonprofits/payments/_details_table.html.erb]: I'm not sure why, but when I was trying to read `payment_details.data.supporter.anonymous` from here it was coming empty, even though this supporter has `false` on its anonymous attribute.

**Review by wwahammy — COMMENTED** [review]
What is the mechanism we use to indicate whether a payment is anonymous or not?

**Comment by clarissalimab** [comment]
@wwahammy on the top of the payment details page we call https://github.com/CommitChange/houdini/blob/supporter_level_goal/app/views/nonprofits/payments/show.rabl#L6 to tell if a payment is anonymous (see https://github.com/CommitChange/houdini/blob/supporter_level_goal/app/models/payment.rb#L52). But since this is the supporter information, I think it should only check if the supporter is anonymous.

**Review by wwahammy — APPROVED** [review]
Normally, supporters are what is marked as anonymous, not the payment. So I guess this does make sense to keep here.

**Review by wwahammy — APPROVED** [review]

**Pull request merged**